pkg/private/util.bzl

```diff
@@ -57,3 +52,28 @@ def setup_output_files(ctx, package_file_name=None):
         output_file = ctx.outputs.out
         output_name = ctx.outputs.out.basename
     return outputs, output_file, output_name
+
+def substitute_package_variables(ctx, attribute_value):
+    """Substitute package_variables in the attribute with the given name.
+
+    Args:
+      ctx: context
+      attribute_value: the name of the attribute to perform package_variables substitution for
+
+    Returns:
+      expanded_attribute_value: new value of the attribute after package_variables substitution
+    """
+    if not ctx.attr.package_variables:
+        # Nothing to substitute. Return the attribute value as is.
+        if type(attribute_value) == "string" and attribute_value.find("{") >= 0:
+            fail("package_variables is required when using '{' in attribute value %s" % attribute_value)
+        return attribute_value
+
+    if not attribute_value:
```

Comment:
I would move this and the next test to the top of the method.
That always catches the error of providing a non-string.
I think the current ordering would silently return a Bool as a value at line 70

Reply:
Checks rearranged.